common.h

```diff
@@ -50,8 +50,9 @@ typedef uint64_t cycles_t;
 
 #define A1HZ 55.
 #define A1MIDI 33
-#define FREQ(x) (65536. / (x))
-#define NOTE(x) (lround(log2(FREQ(x)/A1HZ)*12)+A1MIDI)
+/* wave channel produces half the frequency */
```

Comment:
Technically to be fully correct you'd need to analyze the wave pattern though...

Reply:
zelda.gbs subsong 6 is a nice example where the wave pattern change in the middle of the subsong creates an octave change